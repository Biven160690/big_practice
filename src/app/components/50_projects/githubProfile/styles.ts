import styled, { keyframes, css } from 'styled-components';

export const Input = styled.input`
    width: 100%;
    display: block;
    background-color: #4c2885;
    border: none;
    border-radius: 10px;
    color: #fff;
    padding: 1rem;
    margin-bottom: 5px;
    font-family: inherit;
    font-size: 1rem;
    box-shadow: 0 5px 10px rgba(154, 160, 185, 0.05),
        0 15px 40px rgba(0, 0, 0, 0.1);
`;

export const Block = styled.div`
    width: 300px;
    padding: 10px;
`;

const rotation = keyframes`
0% {
    transform: rotate(0deg);
}
100% {
    transform: rotate(360deg);
}`;

export const Loader = styled.div`
    width: 48px;
    height: 48px;
    border: 5px solid;
    border-color: #ff3d00 transparent;
    border-radius: 50%;
    display: inline-block;
    box-sizing: border-box;
    animation: ${rotation} 1s linear infinite;
`;

const Wrapper = css`
    width: 100%;
    background-color: #4c2885;
    border-radius: 20px;
    box-shadow: 0 5px 10px rgba(154, 160, 185, 0.05),
        0 15px 40px rgba(0, 0, 0, 0.1);
    display: flex;
    padding: 12px;
`;

export const CardWrapper = styled.div`
    ${Wrapper}

    .avatar {
        border-radius: 50%;
        border: 2px solid #2a2a72;
        height: 80px;
        width: 80px;
        margin-right: 10px;
        background: no-repeat center / cover;
    }

    h2 {
        color: #eee;
        margin-bottom: 10px;
    }

    p {
        color: #eee;
        margin-bottom: 10px;
    }

    .followers,
    .following,
    .amountRepos {
        display: flex;
    }

    .amount,
    .text {
        color: #eee;
    }

    .text {
        margin-right: 5px;
    }

    .repos {
        display: flex;
        margin-top: 10px;
    }

    .repo {
        padding: 4px;
        color: #eee;
        text-decoration: none;
        background-color: #212a72;
        
        &:not(:first-child) {
            margin-left: 5px;
        }
    }
`;

export const EmptyCardWrapper = styled.div`
    ${Wrapper}

    justify-content: center;
    align-items: center;

    h2 {
        color: #eee;
    }
`;

export const ErrorCardWrapper = styled.div`
    ${Wrapper}

    justify-content: center;
    align-items: center;

    h1,
    h2 {
        color: #eee;
    }
`;
