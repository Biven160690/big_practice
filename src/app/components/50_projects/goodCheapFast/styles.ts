import styled, { keyframes } from 'styled-components';

const completedToggle = keyframes`
  0% {
    transform: translateX(0px) scale(1);  
}

50% {
    transform: translateX(20px) scale(1.2); 
}
  100% {
    transform: translateX(40px) scale(1);  
    }
`;

const activeToggle = keyframes`
  0% {
    transform: translateX(40px) scale(1);  
}

50% {
    transform: translateX(20px) scale(1.2); 
}
  100% {
    transform: translateX(0px) scale(1);  
}`;

export const Ball = styled.button`
    background: #fff;
    height: 34px;
    width: 34px;
    border-radius: 50%;
    position: absolute;
    top: 3px;
    left: 3px;
    align-items: center;
    justify-content: center;
    border: none;

    &.completed {
        animation: ${completedToggle} 0.3s linear forwards;
    }

    &.active {
        animation: ${activeToggle} 0.3s linear forwards;
    }

    &:hover {
        cursor: pointer;
    }
`;

export const Container = styled.div`
    background-color: #ffffff;
    box-shadow: 0px 2px 10px rgba(255, 255, 255, 0.2);
    padding: 20px;
    width: 350px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
`;

export const Text = styled.div`
    color: #000000;
    font-size: 18px;
    letter-spacing: 1px;
    margin-left: 10px;
`;

export const ToggleArea = styled.div`
    position: relative;
    background-color: #d0d0d0;
    border-radius: 50px;
    cursor: pointer;
    margin: 0 15px 0;
    width: 80px;
    height: 40px;
    transition: background-color 0.3s;

    &.completed {
        background-color: #8e44ad;
    }
`;

export const ToggleContainer = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;

    & + & {
        margin-top: 10px;
    }
`;

export const Block = styled.div`
    display: flex;
    justify-content: center;
    align-items: start;
    flex-direction: column;
    margin-top: 10px;
`