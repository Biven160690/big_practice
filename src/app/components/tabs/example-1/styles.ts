'use client';

import styled, { css } from 'styled-components';

export type TabDirectionProp = {
    position?: string;
};

export type StyledTab = {
    $isActive: boolean;
    color?: string;
    $textSize?: string;
    $bgColor?: string;
    disabled?: boolean;
};

export const StyledTab = styled.button<StyledTab>`
    background-color: ${({ $bgColor = '#244fea', $isActive }) =>
        $isActive ? 'green' : $bgColor};
    border: none;
    color: ${({ color = '#ffffff' }) => color};
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: ${({ $textSize = '16' }) => $textSize}px;
    transition: background-color;
    border-radius: 10px;
    cursor: pointer;

    &:hover {
        filter: brightness(1.3);
    }

    &:disabled {
        filter: brightness(1);
        background-color: #cccccc;
        color: #666666;
        cursor: not-allowed;
    }
`;

const TabDirectionStyles = css`
    width: 100%;
    display: flex;
    margin: 0 15px;
`;

export const TabsRow = styled.div<TabDirectionProp>`
    ${TabDirectionStyles}

    justify-content: ${({ position = 'center' }) => position};
    flex-direction: row;

    & :not(:last-child) {
        margin: 0 10px 0 0;
    }
`;

export const TabsColumn = styled.div<TabDirectionProp>`
    ${TabDirectionStyles}

    align-items: ${({ position = 'center' }) => position};
    flex-direction: column;

    & :not(:last-child) {
        margin: 0 0 10px 0;
    }
`;
