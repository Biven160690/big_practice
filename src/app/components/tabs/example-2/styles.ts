'use client';

import styled from 'styled-components';

export type TabProps = {
    color?: string;
    $textSize?: string;
    $bgColor?: string;
    disabled?: boolean;
}

export type ContainerProps = {
    $currentTab?: number;
    position?: string;
    column?: boolean;
}

export const Tab = styled.button<TabProps>`
    background-color: ${({ $bgColor = '#244fea' }) => ($bgColor)};
    border: none;
    color: ${({ color = '#ffffff' }) => (color)};
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: ${({ $textSize = '16' }) => ($textSize)}px;
    transition: background-color ;
    border-radius:10px;
    cursor: pointer;

    &:hover {
        filter: brightness(1.3);
    }

    &:disabled {
        filter: brightness(1);
        background-color: #cccccc;
        color: #666666;
        cursor: default; 
    }
`;

export const StyledContainer = styled.div<ContainerProps>`
    display: flex;
    justify-content: ${({ position = 'center' }) => (position)};
    flex-direction: ${({ column = false }) => (column ? 'column' : 'row')};
    width: 100%;
    margin: 0 15px;

    & ${Tab}:not(:last-child) {
        margin: ${({ column = false }) => (column ? '0 0 10px 0' : '0 10px 0 0')};
    }

    & ${Tab}:nth-child(${({ $currentTab = 1 }) => $currentTab}) {
        background-color: green;
    }
`;
