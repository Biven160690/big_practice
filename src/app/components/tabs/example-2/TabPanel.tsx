'use client';

import React from 'react';
import { Tab } from './styles';
import { Tabs } from './index';

export const TabPanel = () => {
    const [value, setValue] = React.useState<number>(1);

    const handleChange = (index: number) => setValue(index);

    return (
        <Tabs onChange={handleChange} currentTab={value}>
            <Tab>button </Tab>
            <Tab>button </Tab>
            <Tab>button </Tab>
            <Tab disabled>button</Tab>
        </Tabs>
    );
};
