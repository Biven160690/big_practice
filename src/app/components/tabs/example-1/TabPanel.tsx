'use client';

import React from 'react';

import { Tabs } from './Tabs';
import { Tooltip } from '../../tooltip';
import { Example } from '../../tooltip/example';

export const TabPanel = () => {
    const [currentTab, setCurrentTab] = React.useState<string>('1');

    const handleChange = (value: string) => setCurrentTab(value);

    return (
        <Tabs onChange={handleChange} currentTab={currentTab}>
            <Tabs.Row>
                <Tooltip content={<Example />}>
                    <Tabs.Tab value="1">Tab </Tabs.Tab>
                </Tooltip>
                <Tooltip content={<Example />}>
                    <Tabs.Tab value="2">Tab </Tabs.Tab>
                </Tooltip>
                <Tooltip content={<Example />}>
                    <Tabs.Tab value="3">Tab </Tabs.Tab>
                </Tooltip>
                <Tooltip content={<Example />}>
                    <Tabs.Tab value="4">Tab </Tabs.Tab>
                </Tooltip>
            </Tabs.Row>
        </Tabs>
    );
};
