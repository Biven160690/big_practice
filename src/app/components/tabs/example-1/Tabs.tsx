'use client';

import React from 'react';

import { TabsContext } from './useTabsContext';

import { Tab } from './Tab';
import { TabsColumn, TabsRow } from './styles';

export type EventClick = React.MouseEvent<HTMLButtonElement, MouseEvent>;

export type Props = React.PropsWithChildren<{
    onChange: (value: string, e?: EventClick) => void;
    currentTab: string;
}>;

export const Tabs = ({ children, onChange, currentTab }: Props) => {
    const handleClick = React.useCallback(
        (value: string, e: EventClick, isDisabled: boolean) => {
            if (!isDisabled) {
                onChange(value, e);
            }
        },
        [onChange]
    );

    const value = React.useMemo(
        () => ({ currentTab, handleClick }),
        [currentTab, handleClick]
    );

    return (
        <TabsContext.Provider value={value}>{children}</TabsContext.Provider>
    );
};

Tabs.Tab = Tab;
Tabs.Row = TabsRow;
Tabs.Column = TabsColumn;
