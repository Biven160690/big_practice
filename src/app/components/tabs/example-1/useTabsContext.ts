import React from 'react';

import { EventClick } from './Tabs';

export type TabsPropType = {
    children: React.ReactNode;
};

export type TabsContextType = {
    handleClick: (value: string, e: EventClick, isDisabled: boolean) => void;
    currentTab: string;
};

export const TabsContext = React.createContext<TabsContextType | null>(null);

export function useTabsContext() {
    const context = React.useContext(TabsContext);
    if (!context) {
        throw new Error(
            `Tabs components cannot be rendered outside the TabsProvider`
        );
    }
    return context;
}
