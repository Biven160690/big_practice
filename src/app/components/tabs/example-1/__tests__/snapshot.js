import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { StyledTab, TabsRow, TabsColumn } from '../styles';
import { Tabs } from '../Tabs';
import { TabPanel } from '../TabPanel';
import { TabsContext } from '../useTabsContext';

it('renders StyledTab', () => {
    const tree = renderer.create(<StyledTab />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders TabsRow', () => {
    const tree = renderer.create(<TabsRow />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders TabsColumn', () => {
    const tree = renderer.create(<TabsColumn />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders Tabs', () => {
    const { container } = render(<Tabs />);
    expect(container).toMatchSnapshot();
});

it('renders Tab', () => {
    const tabText = 'Test Tab';
    const value = { currentTab: '1', handleClick: jest.fn() };

    const { container } = render(
        <TabsContext.Provider value={value}>
            <Tabs.Tab value="test" disabled>
                {tabText}
            </Tabs.Tab>
        </TabsContext.Provider>
    );
    expect(container).toMatchSnapshot();
});

it('renders TabPanel', () => {
    const { container } = render(<TabPanel />);
    expect(container).toMatchSnapshot();
});
