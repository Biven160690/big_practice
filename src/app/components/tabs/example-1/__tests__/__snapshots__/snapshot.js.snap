// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StyledTab 1`] = `
.c0 {
  background-color: #244fea;
  border: none;
  color: #ffffff;
  padding: 15px 32px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  -webkit-transition: background-color;
  transition: background-color;
  border-radius: 10px;
  cursor: pointer;
}

.c0:hover {
  -webkit-filter: brightness(1.3);
  filter: brightness(1.3);
}

.c0:disabled {
  -webkit-filter: brightness(1);
  filter: brightness(1);
  background-color: #cccccc;
  color: #666666;
  cursor: not-allowed;
}

<button
  className="c0"
/>
`;

exports[`renders Tab 1`] = `
.c0 {
  background-color: #244fea;
  border: none;
  color: #ffffff;
  padding: 15px 32px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  -webkit-transition: background-color;
  transition: background-color;
  border-radius: 10px;
  cursor: pointer;
}

.c0:hover {
  -webkit-filter: brightness(1.3);
  filter: brightness(1.3);
}

.c0:disabled {
  -webkit-filter: brightness(1);
  filter: brightness(1);
  background-color: #cccccc;
  color: #666666;
  cursor: not-allowed;
}

<div>
  <button
    class="c0"
    disabled=""
  >
    Test Tab
  </button>
</div>
`;

exports[`renders TabPanel 1`] = `
.c1 {
  background-color: green;
  border: none;
  color: #ffffff;
  padding: 15px 32px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  -webkit-transition: background-color;
  transition: background-color;
  border-radius: 10px;
  cursor: pointer;
}

.c1:hover {
  -webkit-filter: brightness(1.3);
  filter: brightness(1.3);
}

.c1:disabled {
  -webkit-filter: brightness(1);
  filter: brightness(1);
  background-color: #cccccc;
  color: #666666;
  cursor: not-allowed;
}

.c2 {
  background-color: #244fea;
  border: none;
  color: #ffffff;
  padding: 15px 32px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  -webkit-transition: background-color;
  transition: background-color;
  border-radius: 10px;
  cursor: pointer;
}

.c2:hover {
  -webkit-filter: brightness(1.3);
  filter: brightness(1.3);
}

.c2:disabled {
  -webkit-filter: brightness(1);
  filter: brightness(1);
  background-color: #cccccc;
  color: #666666;
  cursor: not-allowed;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 :not(:last-child) {
  margin: 0 10px 0 0;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div>
        <button
          class="c1"
        >
          Tab 
        </button>
      </div>
      <div
        class="content"
      >
        <div
          class="example"
        >
          <div
            class="title"
          >
            Hello I am tooltip
          </div>
          <div
            class="text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quidem, atque eligendi eum laudantium illum, a maiores soluta odit doloremque ipsum nostrum error. Fugit magnam sint maiores, vero quae repellendus ad!
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <button
          class="c2"
        >
          Tab 
        </button>
      </div>
      <div
        class="content"
      >
        <div
          class="example"
        >
          <div
            class="title"
          >
            Hello I am tooltip
          </div>
          <div
            class="text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quidem, atque eligendi eum laudantium illum, a maiores soluta odit doloremque ipsum nostrum error. Fugit magnam sint maiores, vero quae repellendus ad!
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <button
          class="c2"
        >
          Tab 
        </button>
      </div>
      <div
        class="content"
      >
        <div
          class="example"
        >
          <div
            class="title"
          >
            Hello I am tooltip
          </div>
          <div
            class="text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quidem, atque eligendi eum laudantium illum, a maiores soluta odit doloremque ipsum nostrum error. Fugit magnam sint maiores, vero quae repellendus ad!
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <button
          class="c2"
        >
          Tab 
        </button>
      </div>
      <div
        class="content"
      >
        <div
          class="example"
        >
          <div
            class="title"
          >
            Hello I am tooltip
          </div>
          <div
            class="text"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quidem, atque eligendi eum laudantium illum, a maiores soluta odit doloremque ipsum nostrum error. Fugit magnam sint maiores, vero quae repellendus ad!
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Tabs 1`] = `<div />`;

exports[`renders TabsColumn 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 15px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 :not(:last-child) {
  margin: 0 0 10px 0;
}

<div
  className="c0"
/>
`;

exports[`renders TabsRow 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 :not(:last-child) {
  margin: 0 10px 0 0;
}

<div
  className="c0"
/>
`;
